### ISSUE TITLE
2.4 requires Jakarta EE 9

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-ldap#639]
This was originally reported [as a Spring Boot issue](https://github.com/spring-projects/spring-boot/issues/30828). It looks like the problem was introduced in [this commit](https://github.com/spring-projects/spring-ldap/commit/28f2ca6ee0a3e87df3afede1ec9aae1a36201197) which accidentally upgraded a number of Jakarta EE dependencies to their EE 9 versions.

Looking at the code, I couldn't find any direct usage of the EE 9 dependencies, other than one in `buildSrc`:

```
$  find . -name "*.java" | xargs grep "import jakarta"
./buildSrc/src/test/resources/samples/integrationtest/withpropdeps/src/integration-test/java/sample/TheTest.java:import jakarta.servlet.http.HttpServletRequest;
```

In addition to correcting the version numbers, perhaps some dependencies could be removed altogether?

**Comment by cmuchinsky** [comment]
I believe there is one here as well:
```
core/src/main/java/org/springframework/ldap/support/LdapEncoder.java:import jakarta.xml.bind.DatatypeConverter
```

**Comment by wilkinsona** [comment]
That's true for the `main` branch but not, AFAICT, for `2.4.x`.

**Comment by jzheaux** [comment]
Closed in 43c57ad50b511b4235de331d33b036976244b05f
